src/crystal/system/unix/file_descriptor.cr

```diff
@@ -26,6 +26,22 @@ module Crystal::System::FileDescriptor
     end
   end
 
+  private def unbuffered_write(*slices : Bytes)
+    writev_syscall_helper(slices.to_a, "Error writing file") do |slices|
```

Comment:
This `to_a` defeats the purpose of the method (it allocates). Does it work without it?

Reply:
I guess we could add `Tuple#static_array`, as `Number#static_array` https://github.com/crystal-lang/crystal/blob/80a975cf0a98a4ce8464d0069145131667becd70/src/number.cr#L65

Reply:
Maybe a splat argument with type like `*slices : Bytes` should just be a `StaticArray`?